### ISSUE TITLE
This is a good idea, but i am not capable of achieving it.

### ISSUE THREAD

**Issue by VigourZHAO** [flameshot-org/flameshot#1487]
**Is your feature request related to a problem?**

**Comment by VigourZHAO** [comment]
1. Add some base flow chart.
    like Excel, like Xmind. When we show it, quickly draw the corresponding flow chart.
2. Support custom graphics.
    Like a custom color.

**Comment by mmahmoudian** [comment]
Thank you for taking time and posting this, but please do not post multiple feature requested in one issue. It would make it hard to track their progress and discussion.

Regarding your first suggestion, flowchart elements, in my opinion that it out of the scope of a screenshot tool. There are better tools to achieve these (and no, Excel is not one of them), such as LibreDraw or yED. You can take the screenshot to those applications and properly create your flowchart.

About your second point, such feature has already been asked. I'll search and reference some of them to this post for your information.

**Comment by thepurple** [comment]
Absolutely agree with @mmahmoudian.

**Comment by VigourZHAO** [comment]
Thank you very much for your reply in your busy schedule.
First of all, According to the first point of view, it will bring a lot of redundancy to this project, I agree with you.
Secondly, What the second point shows. I mean to customize some simple shapes.
Finally, The above content is only personal opinion, if you offend, please forgive me.

**Comment by thepurple** [comment]
No offenses, it's just different opinions.
And thank you for your ideas anyway.

**Comment by mmahmoudian** [comment]
@VigourZHAO
No offence taken. 

Regarding your fist point, I just explain in details the situation and my opinion and for that reason this was only my opinion, I asked the devs of the project to express their opinion as well before we decide to close the issue or take it under consideration.

Regarding your second point, my understanding was that you were looking for some custom colors, but now with your new comment I'm not confident about my initial understanding. Can you elaborate more and explain (perhaps with some examples) what sort of customisation you want to do on simple shapes other than their color.

**Comment by VigourZHAO** [comment]
Currently there are rectangles and circles (quasi-circular)
Other Type: triangles, polygons
Advanced features: Or refer to the custom color function, which supports custom shapes
These are just my thoughts and may not be mature enough. I am very happy to be able to communicate with you guys.

**Comment by mmahmoudian** [comment]
👍🏼  Polygon would be interesting as it gives the user a cleaner way (compared to the Pencil tool) to outline some areas on the screenshot.
🤔 About Triangle I personally have mixed feelings as it is not as useful as a rectangle, plus it has so many different types (Equilateral, Isosceles, Scalene, Right, Right Isosceles, etc.) that adding them would clutter the UI (although in config user can turn on and off certain tools)

➕  I can also add to your idea that we can have:
- different line types
- two headed arrows

This is from the my primary language (R):
![image](https://user-images.githubusercontent.com/390889/111445140-c558ed80-8713-11eb-8cc0-2fef1f02489a.png)
[[source](https://greggilbertlab.sites.ucsc.edu/wp-content/uploads/sites/276/2015/10/linesarrows.jpg)]

❓  How about you create a new issue and propose this idea of yours with the details you just provided.

**Comment by VigourZHAO** [comment]
Haha, if you also find polygons are interesting.(Triangles and rectangles are just a kind of polygons)
I will create a new issue and organize my ideas into a demand book for similar products. But I can't achieve them through code.

**Comment by ZetaoYang** [comment]
@VigourZHAO I suggest you summarize your requirements and change the current issue title.

**Comment by VigourZHAO** [comment]
1、different line types and two headed arrows.
	1. Different dotted lines etc.
2、The drawn graphics support selecting and dragging
	1. If you use the relevant content in the circular circle, if not all of the circle, you can only use Ctrl-z to undo, personally feel inconvenient.
	2. I think the graphics drawn for the first time can be selected and dragged later, which would be better
3、Support vector
	1. That is, select line, click the first point and the second point to form a straight line, and then click the third point, the second point is connected to the third point.
4、Support arc
	1. That is, select arc, Click on the first point, when you click on the second point, drag to display the arc, and so on.
5、Support custom shapes
	1. That is, Google bookmarks, When you draw your commonly used shapes, you can add them through the configuration, which is convenient for later use (similar to the current rectangle and circle).
In summary, the above content is some of my own ideas, and the feasibility of this project needs to be discussed by the participants of this project.

**Comment by VigourZHAO** [comment]
1. different line types and two headed arrows.
   1. Different dotted lines etc.
2. The drawn graphics support selecting and dragging
   1. If you use the relevant content in the circular circle, if not all of the circle, you can only use Ctrl-z to undo, personally feel inconvenient.
   2. I think the graphics drawn for the first time can be selected and dragged later, which would be better
3. Support vector
   1. That is, select line, click the first point and the second point to form a straight line, and then click the third point, the second point is connected to the third point.
4. Support arc
   1. That is, select arc, Click on the first point, when you click on the second point, drag to display the arc, and so on.
5. Support custom shapes
   1. That is, Google bookmarks, When you draw your commonly used shapes, you can add them through the configuration, which is convenient for later use (similar to the current rectangle and circle).

In summary, the above content is some of my own ideas, and the feasibility of this project needs to be discussed by the participants of this project.

**Comment by mmahmoudian** [comment]
The 2nd point is duplicate of #272 and #492.
